blocks/subscriptions-block/components/ContactInfo/index.jsx

```diff
@@ -0,0 +1,164 @@
+import {
+  Button, BUTTON_SIZES, BUTTON_STYLES, FIELD_TYPES, FormInputField, PrimaryFont,
+} from '@wpmedia/shared-styles';
+import React, { useRef } from 'react';
+import { useFusionContext } from 'fusion:context';
+import getProperties from 'fusion:properties';
+import getTranslatedPhrases from 'fusion:intl';
+import countryCodes from './countryCodes';
+import GoogleIcon from './google.svg';
+import FacebookIcon from './facebook.svg';
+import './styles.scss';
+
+const ContactInfo = ({ callback, user, logoutCallback }) => {
+  const formRef = useRef();
+  const entriesRef = useRef({});
+
+  if (user) {
+    entriesRef.current.email = user.email ? user.email : '';
+    entriesRef.current.firstName = user.firstName ? user.firstName : '';
+    entriesRef.current.lastName = user.lastName ? user.lastName : '';
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const valid = formRef.current.checkValidity();
+    if (valid) {
+      callback(entriesRef.current);
+    }
+  };
+
+  const handleInputChange = (name, entry) => {
+    entriesRef.current[name] = entry.value;
+  };
+
+  const handleLogout = () => {
+    logoutCallback();
+  };
+
+  const { arcSite } = useFusionContext();
+  const phrases = getTranslatedPhrases(getProperties(arcSite).locale || 'en');
+
+  const getTranslatedCountries = countryCodes.map((entry) => ({
+    code: entry.code,
+    name: phrases.t(entry.key),
+  }));
+
+  const signOutBtn = (
+    <button
+      type="button"
+      className="sign-out-btn"
+      onClick={handleLogout}
+    >
+      {phrases.t('checkout-block.identity-sign-out')}
+    </button>
+  );
+
+  return (
+    <form onSubmit={handleSubmit} ref={formRef} className="xpmedia-subscriptions-contact-info">
+      <PrimaryFont as="h2" className="xpmedia-subscriptions-contact-info-title">
+        {phrases.t('checkout-block.contact-info')}
+      </PrimaryFont>
+
+      <FormInputField
+        autoComplete="email"
+        label={phrases.t('checkout-block.email')}
+        name="email"
+        defaultValue={user && user.email ? user.email : ''}
+        hidden={!!(user && user.email)}
+        required
+        onChange={(value) => { handleInputChange('email', value); }}
+        showDefaultError={false}
+        type={FIELD_TYPES.EMAIL}
+        validationErrorMessage={phrases.t('checkout-block.email-requirements')}
+      />
+      {user && user.email && (
+        <div className="identity-row">
+            {user?.identities?.[0]?.type && (
+              <>
+                {user.identities[0].type.toLowerCase() === 'google' && (
```

Comment:
We are only ever grabbing the first identity item from the Array. Which means that if the password item is the first item it would always show as them being connected via password even if they use their social login

Reply:
Need more guidance here.  If there are more than one entry in the identity array, what is the preference?

Reply:
Added utility function in Identity hook to address this